aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -888,6 +896,119 @@ module aptos_framework::delegation_pool {
         }
     }
 
+    /// Initialize a delegation pool from an existing owned staking contract.
+    /// The newly created delegation pool will preserve the operator and commission fee. The staker becomes the pool
+    /// owner and has the same delegated voter set.
+    public entry fun initialize_delegation_pool_from_staking_contract(
```

Comment:
When initializing a delegation pool from a staking contract, could you update the code to emit the events we discussed offline?

Reply:
added